### ISSUE TITLE
keda operator metricsserver stuck in ContainerCreating on vcluster

### ISSUE THREAD

**Issue by valkmit** [kedacore/keda#4920]
### Report

Apologies in advance if this is considered out of scope due to the use of vCluster -

keda operator for metrics and admission webhooks pods are stuck in the ContainerCreating state when deployed in a vCluster

I have an EKS cluster provisioned running v1.27. I have provisioned a vCluster within that, with my vCluster version being 0.15.7

Some of the pods are stuck in ContainerCreating due to seeming inability to mount the cert secret

```
$ kubectl get pods -n keda                                    
NAME                                               READY   STATUS              RESTARTS   AGE
keda-operator-7cbf7b677d-cs6m6                     0/1     Pending             0          50m
keda-operator-metrics-apiserver-7c8b69df88-rvp8q   0/1     ContainerCreating   0          50m
keda-admission-webhooks-86d579d9fd-kgkgt           0/1     ContainerCreating   0          50m
```

Below seems to be the issue

```
Events:
  Type     Reason       Age                  From               Message
  ----     ------       ----                 ----               -------
  Normal   Scheduled    46m                  default-scheduler  Successfully assigned keda/keda-operator-metrics-apiserver-7c8b69df88-rvp8q to ip-10-143-161-43.ap-northeast-1.compute.internal
  Warning  FailedMount  16m (x23 over 46m)   kubelet            MountVolume.SetUp failed for volume "certificates" : secret "kedaorg-certs-x-keda-x-devel" not found
  Warning  FailedMount  100s (x20 over 44m)  kubelet            Unable to attach or mount volumes: unmounted volumes=[certificates], unattached volumes=[], failed to process volumes=[]: timed out waiting for the condition

```

There is no secret with that name, and installing keda with helm also did not report any issues.

### Expected Behavior

The containers to come online automatically

### Actual Behavior

The containers did not come online automatically

### Steps to Reproduce the Problem

1.Provision an EKS cluster
2.Create a vCluster workspace and connect to it
3.Install keda within the vCluster

### KEDA Version

None

### Kubernetes Version

1.27

### Platform

Amazon Web Services

**Comment by JorTurFer** [comment]
Hi,
That's totally normal because we introduced certificates and it's the operator who manages them as default (except if you have disabled it because you provided them externally). Your problem is that the operator isn't starting (I see it as pending). Once the operator is ready, it'll create the required secrets for the other pods
